tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java

```diff
@@ -33,56 +31,21 @@
 import org.eclipse.reddeer.swt.impl.shell.DefaultShell;
 import org.eclipse.reddeer.swt.impl.tree.DefaultTree;
 import org.eclipse.reddeer.swt.impl.tree.DefaultTreeItem;
-import org.eclipse.reddeer.workbench.handler.EditorHandler;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
 import org.jboss.ide.eclipse.as.reddeer.server.requirement.ServerRequirement;
+import org.jboss.tools.cdi.bot.test.CDITestBase;
 import org.jboss.tools.cdi.reddeer.cdi.ui.CDIProjectWizard;
 import org.jboss.tools.cdi.reddeer.cdi.ui.wizard.facet.CDIInstallWizardPage;
 import org.jboss.tools.cdi.reddeer.common.model.ui.editor.EditorPartWrapper;
-import org.junit.After;
 import org.junit.Test;
 
-public class CDIWebProjectWizardTemplate{
-	
-	protected String CDIVersion;
-	
-	protected static final String PROJECT_NAME = "CDIProject";
-	protected static final String VERSION = "version";
-	protected static final String FAMILY = "family";
+public class CDIWebProjectWizardTemplate extends CDITestBase {
```

Comment:
You can still use CleanWorkspace requirement for the test class to make sure you got proper setup before test is run.

Reply:
Added.

Reply:
Cool. But... Look at the code, now you have env. prepared before testing, now you should make sure, you clean up after every test method run, if needed of course. 

You do not need to reimplement clean up method, you can call it from parent as you do in other classes, that's ok. 

Then, please, take a look at the other template class, you can use the same process as here, defined requirement to have clean workspace, and then do cleanup in after, instead of before, which is more logical. And If I make comment here only, it does not mean you should not take a look on other places as well, not to wait for me to comment on every place that has the same problem, I can miss it as it is your implementation and you know it better.

Reply:
Fixed.